kubernetes/helm/ckeditor-cs/templates/deployment.yaml

```diff
@@ -0,0 +1,61 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "ckeditor-cs.fullname" . }}
+  labels:
+    {{- include "ckeditor-cs.labels" . | nindent 4 }}
+spec:
+  {{- if not .Values.autoscaling.enabled }}
```

Comment:
https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/#how-does-the-horizontal-pod-autoscaler-work
tl;dr
Autoscaler works as a control loop fetching container metrics (cpu and memory usage) in 15s interval. Then it calculates the percentage of resource usage from comparing values from those metrics against `resources.requests`. That's the default configuration, with some tweaking around it can also work with custom metrics.